src/util/union_find.h

```diff
@@ -168,16 +168,14 @@ class union_find final
   // are 'a' and 'b' in the same set?
   bool same_set(const T &a, const T &b) const
   {
-    number_type na, nb;
-    bool have_na=!numbers.get_number(a, na),
-         have_nb=!numbers.get_number(b, nb);
+    const auto na = numbers.get_number(a);
```

Comment:
Could we do something like naming it `na_opt` so that it is clear that it is an optional?

Reply:
IMO not necessary as it's obvious from context

Reply:
it wasn't so obvious for me: reading `if(na && nb)` makes me think they are `bool` but then on `return uuf.same_set(*na, *nb);` I had to think back.

Reply:
Using `auto` does not always improve readability.

Reply:
I've put explicit types in but kept the names the same